gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/step/util/BulkSetTest.java

```diff
@@ -67,6 +68,56 @@ public void shouldHaveProperCountAndNotOutOfMemoryException() {
         assertEquals(10000000, list.size());
     }
 
+    @Test
+    public void shouldHaveSameClassAsLongAsSameTypesAreAdded() {
+        final BulkSet<Object> bulkSet = new BulkSet<>();
+        bulkSet.add(new ReferenceVertex("1"), 1);
+        bulkSet.add(new ReferenceVertex("2"), 1);
+        bulkSet.add(new ReferenceVertex("3"), 3);
+        assertTrue(bulkSet.allContainedElementsSameClass());
+        assertEquals(bulkSet.getAllContainedElementsClass(), ReferenceVertex.class);
+        bulkSet.add(new ReferenceVertex("4"), 5);
+        assertEquals(bulkSet.getAllContainedElementsClass(), ReferenceVertex.class);
+        bulkSet.add(new Long(2), 5);
+        assertFalse(bulkSet.allContainedElementsSameClass());
+        assertNull(bulkSet.getAllContainedElementsClass());
+    }
+
+    @Test
+    public void shouldNotHaveSameClassForDifferentTypes() {
+        final BulkSet<Object> bulkSet = new BulkSet<>();
+        bulkSet.add(new ReferenceVertex("1"), 1);
+        bulkSet.add(new Long(2), 5);
+        bulkSet.add(new ReferenceVertex("3"), 3);
+        assertNull(bulkSet.getAllContainedElementsClass());
+        assertFalse(bulkSet.allContainedElementsSameClass());
+        bulkSet.add(new ReferenceVertex("4"), 5);
+        assertNull(bulkSet.getAllContainedElementsClass());
+        assertFalse(bulkSet.allContainedElementsSameClass());
+    }
+
+    @Test
+    public void shouldNotHaveSameClassWhenEmpty() {
+        final BulkSet<Object> bulkSet = new BulkSet<>();
+        assertNull(bulkSet.getAllContainedElementsClass());
+        assertFalse(bulkSet.allContainedElementsSameClass());
+    }
+
+    @Test
+    public void shouldNotHaveSameClassForNull() {
```

Comment:
Missing test case for `null` in the middle.
Something like Vertex1, null, Vertex2